src/routes/Profile/Favorites/Favorites.tsx

```diff
@@ -100,23 +113,55 @@ const Favorites = () => {
     }
   }, [selectedFilter, pageData]);
 
-  const favoriteCountText = React.useMemo(() => {
+  const getFavoriteCountsForKaikki = React.useCallback(() => {
+    const jobCount = suosikit.filter((s) => s.tyyppi === 'TYOMAHDOLLISUUS').length;
+    const educationCount = suosikit.filter((s) => s.tyyppi === 'KOULUTUSMAHDOLLISUUS').length;
+    return { jobCount, educationCount };
+  }, [suosikit]);
+
+  const getFavoriteCount = React.useCallback(() => {
+    const { jobCount, educationCount } = getFavoriteCountsForKaikki();
+    if (selectedFilter === 'KAIKKI') {
+      return jobCount + educationCount;
+    }
+    if (selectedFilter === 'TYOMAHDOLLISUUS') {
+      return jobCount;
+    }
+    if (selectedFilter === 'KOULUTUSMAHDOLLISUUS') {
+      return educationCount;
+    }
+    return jobCount + educationCount;
+  }, [getFavoriteCountsForKaikki, selectedFilter]);
+
+  const getFavoriteCountText = React.useMemo(() => {
+    const { jobCount, educationCount } = getFavoriteCountsForKaikki();
     switch (selectedFilter) {
-      case 'KAIKKI':
+      case 'KAIKKI': {
         return t('profile.favorites.you-have-saved-opportunities', {
-          jobOpportunitiesCount: suosikit.filter((s) => s.tyyppi === 'TYOMAHDOLLISUUS').length,
-          educationOpportunitiesCount: suosikit.filter((s) => s.tyyppi === 'KOULUTUSMAHDOLLISUUS').length,
+          jobOpportunitiesCount: jobCount,
+          educationOpportunitiesCount: educationCount,
         });
+      }
       case 'TYOMAHDOLLISUUS':
-        return t('profile.favorites.you-have-saved-n-job-opportunities', {
-          count: totalItems,
+        return t('profile.favorites.you-have-saved-job-opportunities', {
+          jobOpportunitiesCount: jobCount,
         });
       case 'KOULUTUSMAHDOLLISUUS':
-        return t('profile.favorites.you-have-saved-n-education-opportunities', {
-          count: totalItems,
+        return t('profile.favorites.you-have-saved-education-opportunities', {
+          educationOpportunitiesCount: educationCount,
         });
     }
-  }, [selectedFilter, suosikit, t, totalItems]);
+  }, [getFavoriteCountsForKaikki, selectedFilter, t]);
+
+  const getFilterValueInTools = React.useMemo(() => {
```

Comment:
Maybe `getFilterValueForTools` would be more describing name, as we are not fetching the current status from Tools but instead were are going to give it.